src/tscore/MatcherUtils.cc

```diff
@@ -618,3 +618,75 @@ parseConfigLine(char *line, matcher_line *p_line, const matcher_tags *tags)
 
   return nullptr;
 }
+
+// char* parseYamlDoc(YAML::Node node, matcher_line* p_line, const matcher_tags* tags)
+//
+//   Parse out a yaml node object suitable for passing to
+//    a ControlMatcher object
+//
+//   If successful, nullptr is returned.  If unsuccessful,
+//     a static error string is returned
+//
+const char *
+parseYamlDoc(matcher_line *p, const matcher_tags *tags)
+{
+  const char *label;
+  const char *val;
+  matcher_type type = MATCH_NONE;
+
+  YAML::Node root = *(p->root_node);
+  YAML::Node node = root[p->line_num];
+  uint dest       = 0;
+  if (node[tags->match_ip]) {
+    type  = MATCH_IP;
+    label = tags->match_ip;
+    val   = node[tags->match_ip].Scalar().c_str();
```

Comment:
Might as well make `val` a `std::string_view`.

Reply:
changed val to a string_view.